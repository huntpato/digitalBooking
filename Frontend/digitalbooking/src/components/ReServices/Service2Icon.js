import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styles from './Service2Icon.module.css'
import {
  faBan,
  faWifi,
  faSwimmer,
  faTv,
  faUtensils,
  faPaw,
  faCar,
  faTemperatureLow,
  faSmoking,
  faHotTub,
  faCoffee, 
  faSnowflake,
  faMugHot,
  faDumbbell,
  faWheelchair,
} from "@fortawesome/free-solid-svg-icons";

const dict = {
  wifi: faWifi,
  pool: faSwimmer,
  cabletv: faTv,
  kitchen: faUtensils,
  petsallowed: faPaw,
  parking: faCar,
  airconditioning: faTemperatureLow,
  smokingallowed: faSmoking,
  jacuzzi: faHotTub,
  breakfast: faCoffee,
  refrigerator: faSnowflake,
  microwave: faMugHot,
  gym: faDumbbell,
  accesible : faWheelchair
};

const Service2Icon = ({ icon, color }) => {
  let service;

  if (icon === undefined && !dict.hasOwnProperty(icon)) {
    service = faBan;
  } else {
    service = dict[icon.toLowerCase().replace(/ /g, "")];
  }

  return (
    <>
      <FontAwesomeIcon className= {icon} icon={service} style={color} />
    </>
  );
};

export default Service2Icon;
