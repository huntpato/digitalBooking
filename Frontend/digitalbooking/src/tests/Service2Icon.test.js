import React from 'react';
import Service2Icon from "../components/ReServices/Service2Icon";
import { expect } from '@jest/globals';
import '@testing-library/jest-dom'

describe('Comprobar funcion Service2Icon', ()=> {


    test('Verificar que la funcion se ejecuta correctamente', () => {
        const services = [
            "Wifi"
        ];
        expect(Service2Icon(services)).toMatchSnapshot();;

})
})