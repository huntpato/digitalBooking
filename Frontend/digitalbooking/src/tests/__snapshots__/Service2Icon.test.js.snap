// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comprobar funcion Service2Icon Verificar que la funcion se ejecuta correctamente 1`] = `
<React.Fragment>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f05e",
          "M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z",
        ],
        "iconName": "ban",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</React.Fragment>
`;
